src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,29 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, CompletableFuture<UserData>> cache = new ConcurrentHashMap<>();
```

Comment:
In this task we use `Future` to represent that we may have to go to the DB to get the data but we may also already have it in cache and return it immediately

This map should store already "fetched" user data, `CompletableFuture` is a wrapper class that stores the result of some operations that may be completed already, but also maybe not yet.
```suggestion
    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
```